lib/Runtime/Library/GlobalObject.cpp

```diff
@@ -2175,27 +2175,28 @@ namespace Js
         return TRUE;
     }
 
-    BOOL GlobalObject::HasItem(uint32 index)
+    PropertyQueryFlags GlobalObject::HasItemQuery(uint32 index)
     {
-        return DynamicObject::HasItem(index)
+        return JavascriptConversion::BooleanToPropertyQueryFlags(JavascriptConversion::PropertyQueryFlagsToBoolean((DynamicObject::HasItemQuery(index)))
             || (this->directHostObject && JavascriptOperators::HasItem(this->directHostObject, index))
-            || (this->hostObject && JavascriptOperators::HasItem(this->hostObject, index));
+            || (this->hostObject && JavascriptOperators::HasItem(this->hostObject, index)));
     }
 
     BOOL GlobalObject::HasOwnItem(uint32 index)
     {
-        return DynamicObject::HasItem(index)
+        return JavascriptConversion::PropertyQueryFlagsToBoolean(DynamicObject::HasItemQuery(index))
             || (this->directHostObject && this->directHostObject->HasItem(index));
     }
 
-    BOOL GlobalObject::GetItemReference(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
+    PropertyQueryFlags GlobalObject::GetItemReferenceQuery(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
     {
-        if (DynamicObject::GetItemReference(originalInstance, index, value, requestContext))
+        if (JavascriptConversion::PropertyQueryFlagsToBoolean(DynamicObject::GetItemReferenceQuery(originalInstance, index, value, requestContext)))
         {
-            return TRUE;
+            return Property_Found;
         }
-        return (this->directHostObject && this->directHostObject->GetItemReference(originalInstance, index, value, requestContext)) ||
-            (this->hostObject && this->hostObject->GetItemReference(originalInstance, index, value, requestContext));
+        return JavascriptConversion::BooleanToPropertyQueryFlags(
+            (this->directHostObject && JavascriptConversion::PropertyQueryFlagsToBoolean(this->directHostObject->GetItemReferenceQuery(originalInstance, index, value, requestContext))) ||
```

Comment:
Yes, because the definition of JavascriptConversion::PropertyQueryFlagsToBoolean() can be easily changed in the future without touching all these sites.


---
In reply to: [106305716](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106305716) [](ancestors = 106305716)